src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
> Only say the word and it shall be done.

Consider it uttered on grounds of the non-short tail of interactions with weak subtyping and the expanded horizons that `42.type` promises.

You can keep the `if (isScala213)` guard as it currently stands as `scalac213 -Xsource:2.12` is a supported thing, insofar as we remember and are able to maintain the historical behaviour.